src/encoding/codecs/vpXfamily.md

```diff
@@ -0,0 +1,292 @@
+(realtime video for example)) respectively.
+
+##### `--deadline best`
+
+this instructs the encoder to be more strict about the quality of
+the encoding and not to move on until it is 100% satisfied with
+that frame.
+
+
+##### `--auto-alt-ref=2`
+
+This is a option that determines how keyframes are chosen. If you
+are tuning for maximum playback compatability choose 1, if you are
+tuning for maximum quality choose 2.
+
+
+##### `--lag-in-frames=25`
+
+This option allows keyframes to be moved forward or back by this
+number of frames which allows for better keyframe placement.
+larger numbers like 25 are better suited for anime.
+
+
+##### `--arnr-strength=0`
+
+Turn off keyframe denoising as you've already pre-processed the
+video with vapoursynth.
+
+
+##### `--aq-mode=2`
+
+This option tells the encoder to rely on how complex the motion
+of each frame is to determine the adaptive quantization. In
+other words, more motion in more directions equals more bits
+devoted to that frame.
+
+
+##### `--frame-parallel=0`
+
+This option allows for parallel decoding of the video, it reduces
+quality so we disable it.
+
+
+##### `--tile-columns=2` and `--tile-rows=1`
+
+These options determine how the encoder breaks up the frame into
+chunks for encoding. The more chunks the more threads can be used
+for encoding. One important note here is that the values of this
+parameter are in the form of $$2^{x}$$ meaning that the number of
+columns and rows with these values is actually 4 and 2 
+respectively. Tiles cannot be bigger than 65536 pixels in area.
+
+
+##### `--threads=8`
+
+Self explanitory. Set this to the number of threads your cpu has.
+
+
+##### `--end-usage=vbr`
+
+Set the encoder in VBR mode.
+
+
+##### `--cq-level=25`
+
+Set the average quantizer level.
+
+
+##### `-o /dev/null temp.y4m` or `-o myvideo.ivf temp.y4m`
+
+This last portion tells which files to use for the input and output.
+We use `-o` to tell which filename to write the encoded file to.
+In the first case, vpxenc will write a file at `/dev/null` which is
+the null device on posix systems (meaning no file will be written
+for the first pass (if you're on windows simply use the filename nul
+instead)). In the second pass we write an ivf to `myvideo.ivf` in the
+current working directory.
+
+The last argument we are passing to vpxenc is the input file.
+As we are doing a two pass encode it is more efficient to save the
+output of vapoursynth then run the encode on that instead of running
+vapoursynth twice.
+
+
+### Example 2: General-Purpose Encoding with AV1
+
+TODO
+
+## Recap
+
+We covered the basics of how to encode using the vpX/aom family of
+codecs, including speed presets.
+
+Here is a summary of when to use vp9 vs AV1:
+
+- Is your video larger than Full HD?
+	- Yes:
+		- AV1
+	- No:
+		- Is your source Full HD Quality:
+			- Yes:
+				- Is time important?
+					- Yes:
+						- VP9
+					- No:
+						- AV1
+			- No:
+				- VP9
```

Comment:
Why? AV1 can still achieve better compression, so I would argue a source being low quality should not be a reason to use VP9 over AV1.

Reply:
AV1 Is inefficient for files that have a lower resolution than 1080p due to
the way that the tiles are handled. That being said this is just a
recommendation and you can encode lower res with AV1.

Reply:
Fair, although there's no need to use tiles. You can get efficient multithreading using a tool like av1an, and set tiles to 1.

Reply:
Split encoding (like how av1an works) actually results in higher bitrates compared with tile multithreading.

Reply:
Do you have any research to support that? Logically, there's no reason that split encoding should result in higher bitrate than using tiles. It should be the opposite. A split encoded stream should be the equivalent of encoding that stream as one sequence.*

*Assuming QP based encoding. Don't do split encoding with VBR.

Reply:
Think about it this way, each segment must start with a keyframe, If a keyframe isn't necessary in that location it's going to increase bitrate. Even if it is, among other things, libaom-av1 optimizes keyframe placement to occur on frames with large changes to the image. This means that the keyframe will likely be placed sub-optimally if you split-encode it, increasing the bitrate or lowering the quality. libaom-av1 has some very complex keyframe optimization techniques it uses which can be thrown off by split encoding like that. libvpx-vp9 does some similar but less-complex optimizations, so it won't be affected as much but it still can be.

Reply:
But anyways... Is there anything else I need to fix?